docs/docs/architecture.md

```diff
@@ -1,143 +1,148 @@
-constraint may specify that a specific set of code hashes is required i.e. every version is whitelisted and no
-upgrade is possible until the constraint is adjusted. Or, it may specify a set of allowed signing keys, enabling
-enclave authors to release new versions whenever they want. In that scenario the enclave creator is trusted, 
-but the entity hosting the enclave instance may not be.
-
-When they're happy, they create encrypted messages using the key in the [`EnclaveInstanceInfo`](api/-conclave%20-core/com.r3.conclave.common/-enclave-instance-info/index.html). By sending and
-receiving such messages to the host (and from there to the enclave), communication is established. See the
-[Mail](#mail) section below for further discussion of this.
-
-Whilst the high level setup has just those three entities, real deployments have more:
-
-1. Intel
-2. Optionally, a cloud provider
-3. Optionally, an auditor
-
-**Intel.** Intel's involvement in a deployed architecture is limited to providing the CPU hardware and running some servers.
-These servers provision the host with certificates guaranteeing authenticity of the chip (once, on initial setup)
-and provide the client with an assessment of the security of the host given a remote attestation (whenever the
-client requests it). If Intel's servers become unreachable by the clients that's not a problem, it just means the
-security of the host machine may degrade without the clients realising.
-
-For instance if the host's SGX software stack is out of date and has known vulnerabilities, or if the BIOS configuration is
-not correct, this will be reported to the client by Intel's servers as part of verifying and processing the data inside
-the remote attestation provided by the host.
-
-**Cloud provider.** A cloud provider needs to support SGX for Conclave to be usable. They may operate their own
-provisioning servers that take over from Intel's. You can find the instructions to use Conclave on Microsoft Azure
-[here](machine-setup.md).
-
-**Auditor.** In the pure enclave-oriented model, the user is responsible for understanding what the enclave does before
-using it by reading the enclave's source code. If the user doesn't do this then enclaves have no point. In practice the
-user may wish to outsource this auditing to someone else.
+Clients need to be sure that they are communicating with a real enclave. They cannot rely on the host to _assert_ 
+that this is the case because the host could be malicious. Instead, Conclave uses a process called
+*remote attestation*.
+
+First, clients need to obtain an
+[`EnclaveInstanceInfo`](api/-conclave%20-core/com.r3.conclave.common/-enclave-instance-info/index.html) object. You 
+can download this object on-demand from the host, or it could be published somewhere. 
+
+The `EnclaveInstanceInfo` object encapsulates a remote attestation. The remote attestation includes the following 
```

Comment:
This reference to `EnclaveInstanceInfo` and the two below do not link to the JavaDoc, whilst other references do. Do we need a consistent scheme for this? I would prefer all references to public API classes to refer to the JavaDoc as a link.

Reply:
Done. I will update the style guide too.